### PR TITLE
Ddcnl 3914

### PR DESCRIPTION
**Pull request by Tateni93**

### CODE REVIEW

File: test/services/RequestDetailsServiceSpec.scala [hunk 1]
```diff
@@ -48,16 +48,16 @@ class RequestDetailsServiceSpec extends BaseSpec {
...
-        SUT.validateDates(requestDetails) mustBe Left(Constants.responseInvalidPayload)
-      }
-    }
+//    "return responseInvalidPayload" when {
+//      "the dates are not parsable" in {
+//        val requestDetails = createRequestDetails("202-04-1", "122-01-01")
+//
+//        SUT.validateDates(requestDetails) mustBe Left(Constants.responseInvalidPayload)
+//      }
+//    }
```
Comment (pascal22p) [inline test/services/RequestDetailsServiceSpec.scala:57]: Forgot something?

**Review by pascal22p — APPROVED** [review]

File: test/services/RequestDetailsServiceSpec.scala (hunk 1 above)
Comment (Tateni93) [reply test/services/RequestDetailsServiceSpec.scala]: Removed as validation moved to the case class

File: app/connectors/DesConnector.scala [hunk 2]
```diff
@@ -34,15 +34,14 @@ class DesConnector @Inject() (httpClient: HttpClient, desConfig: ApplicationConf
...
-      Seq(
-        "Authorization" -> desConfig.authorization,
-        "Environment"   -> desConfig.environment,
-        "CorrelationId" -> correlationID
-      )
+  private def header(correlationID: String): Seq[(String, String)] =
+    Seq(
+      "Authorization" -> desConfig.authorization,
+      "Environment"   -> desConfig.environment,
+      "CorrelationId" -> correlationID
```
Comment (pascal22p) [inline app/connectors/DesConnector.scala:41]: Sorry, going to be annoying again, can you do the same as tamc:
def explicitHeaders(implicit hc: HeaderCarrier): List[(String, String)] =
    List(
      HeaderNames.authorisation -> urlHeaderAuthorization,
      HeaderNames.xRequestId    -> hc.requestId.fold("-")(_.value),
      HeaderNames.xSessionId    -> hc.sessionId.fold("-")(_.value),
      "Environment"             -> urlHeaderEnvironment,
      "CorrelationId"           -> UUID.randomUUID().toString)

https://github.com/hmrc/tamc/pull/56/files

For the correlationId, just use a randomUUID. You cannot trust the value from the user to be unique and in fact I know that DWP reuse the same id on failures...

**Review by pascal22p — APPROVED** [review]

File: app/connectors/DesConnector.scala (hunk 2 above)
Comment (Tateni93) [reply app/connectors/DesConnector.scala]: Headers added

File: app/connectors/DesConnector.scala [hunk 3]
```diff
@@ -34,15 +34,14 @@ class DesConnector @Inject() (httpClient: HttpClient, desConfig: ApplicationConf
...
-  private def header(correlationID: String): HeaderCarrier =
-    HeaderCarrier(extraHeaders =
-      Seq(
-        "Authorization" -> desConfig.authorization,
-        "Environment"   -> desConfig.environment,
-        "CorrelationId" -> correlationID
-      )
+  private def header(correlationID: String): Seq[(String, String)] =
+    Seq(
+      "Authorization" -> desConfig.authorization,
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply app/connectors/DesConnector.scala]: Headers added

**Review by pascal22p — APPROVED** [review]

**Review by kurtlogan — APPROVED** [review]

**Pull request merged**